mvpa2/datasets/eventrelated.py

```diff
@@ -641,3 +644,56 @@ def eventrelated_dataset(ds, events, time_attr=None, match='prev',
                     regr_attrs=regr_attrs)
     else:
         raise ValueError("unknown event model '%s'" % model)
+
+
+def get_contrasts(hrf_estimates, contrasts=None,
+                  condition_attr='targets',
+                  fxname="z_score"):
+    """A helper to obtain contrasts from the fit NiPy GLM model
+
+    .a.model
+
+    Parameters
+    ----------
+    hrf_estimates: Dataset
+      Output from `fit_event_hrf_model` ran with return_model=True
+    contrasts: dict, optional
+      name: dict of coefficients per condition. If None, value is returned for
+      each condition
+    """
+    conditions = hrf_estimates.sa[condition_attr].value
+    model = hrf_estimates.a.model
+    # additional regressors which were added to the model by us (e.g. "const")
+    # but estimates for which are not present
+    nadd_regs = len(hrf_estimates.a.add_regs.sa.regressor_names)
+
+    if contrasts is None:
+        contrasts = OrderedDict(
+            (c, {c: 1})
+            for c in conditions
+        )
+
+    out = Dataset(
+        np.empty((len(contrasts), hrf_estimates.nfeatures)),
+        "TODO"
+    )
+    out.sa.update(hrf_estimates.sa)
+    out.fa.update(hrf_estimates.fa)
+    for i, contrast in enumerate(contrasts):
+        # convert our presentation for contrasts into NiPy's
+        for c, v in contrast.items():
+            condition_index = [conditions == c]
+            if not any(condition_index):
+                raise ValueError(
+                    "Found no condition %s among available: %s  in the contrast %s"
+                    % (c, conditions, contrast)
+                )
+            assert sum(condition_index) == 1
+        # and add complementary regressors
```

Comment:
Hi @yarikoptic, I'm taking a look to understand it atm (basically, going through everything that doesn't have a TODO in it in ipython, Python 2.7.16rc1), and currently, I'm running into problems in this for-loop. I might just miss something, but wanted to share. 


<details>
<summary>Background:</summary>

 ``hrf_estimates`` is computed with 
```python
    hrf_estimates = mv.fit_event_hrf_model(mean_sens_transposed,
                                           events_dicts,
                                           time_attr='time_coords',
                                           condition_attr='condition',
                                           design_kwargs=dict(drift_model='blank'),
                                           glmfit_kwargs=dict(model='ols'),
                                           return_model=True)
```
``conditions = hrf_estimates.sa[condition_attr].value``  in the example data I'm playing around with (localizer) is

```python
In [75]: conditions
Out[75]: 
array(['body', 'body_first', 'face', 'face_first', 'house', 'house_first',
       'object', 'object_first', 'scene', 'scene_first', 'scramble',
       'scramble_first'], dtype='|S14')
```
</details>


```python
if contrasts is None:
        contrasts = OrderedDict(
            (c, {c: 1})
            for c in conditions
        )
```

yields

```python
In [71]: contrasts
Out[71]: 
OrderedDict([('body', {'body': 1}),
             ('body_first', {'body_first': 1}),
             ('face', {'face': 1}),
             ('face_first', {'face_first': 1}),
             ('house', {'house': 1}),
             ('house_first', {'house_first': 1}),
             ('object', {'object': 1}),
             ('object_first', {'object_first': 1}),
             ('scene', {'scene': 1}),
             ('scene_first', {'scene_first': 1}),
             ('scramble', {'scramble': 1}),
             ('scramble_first', {'scramble_first': 1})])
```

But in the for loop throws a bunch of subsequent errors. The first one related to the fact that enumerating contrasts does not return a dict in ``contrast`` for me, thus leading to

```python
In [87]:     for i, contrast in enumerate(contrasts):
    ...:         # convert our presentation for contrasts into NiPy's
    ...:         for c, v in contrast.items():
    ...:             condition_index = [conditions == c]
    ...:             if not any(condition_index):
    ...:                 raise ValueError(
    ...:                     "Found no condition %s among available: %s  in the contrast %s"
    ...:                     % (c, conditions, contrast)
    ...:                 )
    ...:             assert sum(condition_index) == 1
    ...:         
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-87-5e7e793a0b02> in <module>()
      1 for i, contrast in enumerate(contrasts):
      2     # convert our presentation for contrasts into NiPy's
----> 3     for c, v in contrast.items():
      4         condition_index = [conditions == c]
      5         if not any(condition_index):

AttributeError: 'numpy.string_' object has no attribute 'items'

```

Also, ``condition_index = [conditions == c]`` is a list, thus ``any(condition_index)`` and ``sum(condition_index)`` throw errors. For me, the following changes make everything run:

```python 
    for i, contrast in enumerate(contrasts):
              # convert our presentation for contrasts into NiPy's
              for c, v in contrasts[contrast].items():
                  condition_index = [conditions == c]
                  if not any(condition_index[0]):
                      raise ValueError(
                          "Found no condition %s among available: %s  in the contrast %s"
                          % (c, conditions, contrast)
                      )
                  assert sum(condition_index[0]) == 1

```

Reply:
Thank you for checking out!! That was just a stub at a solution, I've not even tried to run it -- just was composing the logical blocks I saw needing to exist.  So it is expected that I screwed up in that loop ;)  use `np.any(condition_index)` instead of `any(...)` and it should be not `_index` but `_mask` I guess now ;) etc.  You are most welcome to complete it!

Reply:
I think this one was resolved